Refactor docs struct and content. (#69)
streamnative/pulsar-io-aws-eventbridge: docs/aws-eventbridge-sink.md

![](/docs/aws-eventbridge-sink.png)

## Quick start

### Prerequisites

Prerequisites for an AWS EventBridge sink connector connecting to external systems:

1. Create EventBridge and EventBus in AWS.

> 2. Create the [AWS User](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) and create `Access Key`(Please record `accessKey` and `secretAccessKey`).

Reviewer: Any difference between `Access Key` and `accessKey`? Does it make sense to get them aligned?

> 3. Set permissions to users, ensure that the user has the following permissions to the AWS EventBus. For details, see [permissions for event buses](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-bus-perms.html)

Reviewer: ```suggestion
3. Assign permissions to Pulsar users, and ensure they have the following permissions to the AWS EventBus. For details, see [permissions for event buses](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-event-bus-perms.html)
```

![](/docs/aws-eventbridge-sink.png)

## Quick start

### Prerequisites

Prerequisites for an AWS EventBridge sink connector connecting to external systems:

1. Create EventBridge and EventBus in AWS.
2. Create the AWS authentication key, which supports two authentication methods:

> - [Access Keys](https://docs.aws.amazon.com/powershell/latest/userguide/pstools-appendix-sign-up.html): If you use this method, Just need set `accessKeyId` and `secretAccessKey` on this connector config.

Reviewer: ```suggestion
- [Access Keys](https://docs.aws.amazon.com/powershell/latest/userguide/pstools-appendix-sign-up.html): To use this method, you need to set `accessKeyId` and `secretAccessKey` on this connector config.
```

> - [Security Token Service](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html): If you use this method, Just need set `stsEndpoint`, `stsRegion`, `role`, and `roleSessionName` on this connector config.

Reviewer: ```suggestion
- [Security Token Service](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html): To use this method, you need to set `stsEndpoint`, `stsRegion`, `role`, and `roleSessionName` on this connector config.
```

```json
  {
    "Version": "2012-10-17",
    "Statement": [

      ### To grant permissions for an account to use the PutEvents action, include the following, otherwise delete this section: ###

      {

        "Sid": "AllowAccountToPutEvents",
        "Effect": "Allow",
        "Principal": {
          "AWS": "<ACCOUNT_ID>"
        },
        "Action": "events:PutEvents",
        "Resource": "{EventBusArn}"
      },

      ### Include the following section to grant permissions to all members of your AWS Organizations to use the PutEvents action ###

      {
        "Sid": "AllowAllAccountsFromOrganizationToPutEvents",
        "Effect": "Allow",
        "Principal": "*",
        "Action": "events:PutEvents",
        "Resource": "{EventBusArn}"
        "Condition": {
          "StringEquals": {
            "aws:PrincipalOrgID": "o-yourOrgID"
          }
        }
      }
    ]
}
```
4. Create a [Rule](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-create-rule.html) in EventBridge.
- The data structure sent to Event Bridge is described [here](## Metadata mapping), and you can create **event pattern** based on this structure.

> - Set the target according to your needs. If you're testing this connector, you can set target to [Cloud Watch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html)

Reviewer: ```suggestion
- Set the target according to your needs. If you're testing this connector, you can set the target to [Cloud Watch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html).
```

> ### 1. Create Connector

Reviewer: ```suggestion
### 1. Create a connector
```

> To create AWS EventBridge sink connector, depending on the environment, there are several ways to create it:

Reviewer: ```suggestion
Depending on the environment, there are several ways to create an AWS EventBridge sink connector:
```

- [Create Connector on StreamNative Cloud](https://docs.streamnative.io/docs/connector-create).
- [Create Connector with Function worker](https://pulsar.apache.org/docs/3.0.x/io-quickstart/). 

>   This way need download **nar** package to create build-in or no-build-in connector. You can download the version you need from [here](https://github.com/streamnative/pulsar-io-aws-eventbridge/releases)

Reviewer: ```suggestion
  Using this way requires you to download a **NAR** package to create a built-in or non-built-in connector. You can download the version you need from [here](https://github.com/streamnative/pulsar-io-aws-eventbridge/releases).
```

- [Create Connector with Function mesh](https://functionmesh.io/docs/connectors/run-connector).

>   This way need set docker image. You can choose the version you want to launch from [here](https://hub.docker.com/r/streamnative/pulsar-io-aws-eventbridge/tags)

Reviewer: ```suggestion
  Using this way requires you to set the docker image. You can choose the version you want to launch from [here](https://hub.docker.com/r/streamnative/pulsar-io-aws-eventbridge/tags)
```

> Please go to the corresponding document for details. 

Reviewer: We can remove this sentence. It's explicit.

> Each creation way is similar, you just need to set the following parameters to the connector configuration.

Reviewer: ```suggestion
No matter how you create an AWS EventBridge sink connector, the minimum configuration contains the following parameters.
```

```json
   "configs": {
      "accessKeyId": "{{Your access access key}}",
      "secretAccessKey": "{{Your secret access key}}",
      "region": "test-region",
      "eventBusName": "test-event-bus-name"
   }
}
```

> > * The above different creation methods are different for the structure of the configuration, 

Reviewer: ```suggestion
> * The configuration structure varies depending on how you create the AWS EventBridge sink connector.
```

> >  some are **json**, some are **yaml**, some are **k8s yaml**, you only need to replace with the corresponding format.

Reviewer: ```suggestion
>  For example, some are **JSON**, some are **YAML**, and some are **Kubernetes YAML**. You need to adapt the configs to the corresponding format.
```

>  
> * If you want to configure more, please refer to **Configuration Properties**

> ### 2. Send message to topic

Reviewer: ```suggestion
### 2. Send messages to the topic
```

> > If your connector is created on StreamNative Cloud, please refer to this [document](TODO) on how to authenticate clients.

Reviewer: ```suggestion
> **Note:** If your connector is created on StreamNative Cloud, you need to authenticate your clients. See [Build applications using Pulsar clients](https://docs.streamnative.io/docs/qs-connect#jumpstart-for-beginners) for more information.
```

            .serviceUrl("{{Your Pulsar URL}}")
            .build();

        Producer<String> producer = client.newProducer(Schema.STRING)
            .topic("{{Your topic name}}")
            .create();

        String message = "{\"msg\": \"msg-data\"}";
        MessageId msgID = producer.send(message);
        System.out.println("Publish " + message + " and message ID " + msgID);
        
        producer.flush();
        producer.close();
        client.close();
```

### 3. Show data on AWS EventBridge
The connector will send the following format of JSON event to EventBridge.
```json
{
  "version": "0",
  "id": "6a7e8feb-b491-4cf7-a9f1-bf3703467718",
  "detail-type": "{{Your topic name}}",
  "source": "{{Your connector name}}",
  "account": "111122223333",
  "time": "2017-12-22T18:43:48Z",
  "region": "us-west-1",
  "resources": [
    "arn:aws:ec2:us-west-1:123456789012:instance/i-1234567890abcdef0"
  ],
  "detail": {
    "data": {
      "msg": "msg-data"
    },
    "message_id": "124:191:0"
  }
}
```

## Configuration Properties

Before using the AWS EventBridge sink connector, you need to configure it. This table outlines the properties and the
descriptions.

> This connector provider two authentication methods for AWS EventBridge, please select either authentication method from `Access keys` and `Security Token Service`.

Reviewer: ```suggestion
This connector provides two authentication methods for AWS EventBridge. You can select either one between `Access keys` and `Security Token Service`.
```

| `stsEndpoint`     | String | yes      | "" (empty string) | The AWS STS endpoint. By default, the default STS endpoint: https://sts.amazonaws.com is used. See [Amazon documentation](https://docs.aws.amazon.com/STS/latest/APIReference/welcome.html) for more details. |
| `stsRegion`       | String | yes      | "" (empty string) | The AWS STS region, By default, the 'region' config or env region is used.                                                                                                                                    |
| `role`            | String | yes      | "" (empty string) | The AWS STS [role.](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)                                                                                                                  |
| `roleSessionName` | String | yes      | "" (empty string) | The AWS STS [role session name.](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html)                                                                                                     |

> - Other Configuration

Reviewer: ```suggestion
- Other configurations
```

| `batchMaxTimeMs`        | long   | no       | 5000              | Batch max wait time: milliseconds.                                                                                                                                                                                                       |
| `maxRetryCount`         | long   | no       | 100               | Maximum number of retries to send events, when put events failed.                                                                                                                                                                        |
| `intervalRetryTimeMs`   | long   | no       | 1000              | The interval time(milliseconds) for each retry, when the put events failed.                                                                                                                                                              |

> > For advanced features and configuration of this connector, please refer to the Advanced feature for more information.

Reviewer: ```suggestion
> For details about this connector's advanced features and configurations, see [Advanced features](#advanced-features).
```

> ## Advanced feature

Reviewer: ```suggestion
## Advanced features
```

intervalRetryTimeMs: 1000 //The interval time(milliseconds) for each retry, when event send failed.
```

> ## More Link

Reviewer: ```suggestion
## More links
```
